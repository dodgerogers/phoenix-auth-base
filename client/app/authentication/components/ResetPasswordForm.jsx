import React from 'react'
import { Field } from 'redux-form/immutable';
import { Form, Message, Button, Icon } from 'semantic-ui-react';
import Input from '../../common/components/Input';
import { isEmail, isRequired, minLength, matchField } from '../../lib/utils/validation';

const minPasswordLength = minLength(8);
const passwordsMustMatch = matchField('password');


const ResetPasswordForm = (props) => {
  const { handleSubmit, error, submitting, dirty } = props;

  return (
    <div className="reset-password-form">
      <Form
        size='large'
        loading={submitting}
        onSubmit={handleSubmit}
      >
        {error && dirty && <Message negative>{error}</Message>}
        <Field
          component={Input}
          fluid={true}
          name="email"
          icon="mail"
          iconPosition="left"
          placeholder="Email"
          validate={[isRequired, isEmail]}
        />
        <Field
          component={Input}
          fluid={true}
          name="resetPasswordToken"
          placeholder="Reset Password Token"
          validate={[isRequired]}
        />
        <Field
          component={Input}
          fluid={true}
          name="password"
          icon="lock"
          type="password"
          iconPosition="left"
          placeholder="New Password"
          validate={[isRequired, minPasswordLength]}
        />
        <Field
          component={Input}
          fluid={true}
          name="passwordConfirmation"
          type="password"
          icon="lock"
          iconPosition="left"
          placeholder="Password Confirmation"
          validate={[isRequired, passwordsMustMatch]}
        />
        <Button
          type="submit"
          fluid={true}
          color="teal"
          disabled={!props.valid}
        >
          Submit
        </Button>
      </Form>
    </div>
  );
}

export default ResetPasswordForm;
