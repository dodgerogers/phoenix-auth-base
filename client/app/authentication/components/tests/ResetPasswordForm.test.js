import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ResetPasswordForm from '../ResetPasswordForm';


describe('ResetPasswordForm', () => {
  const fullProps = {};

  const pureComponent = (props) => shallow(
    <ResetPasswordForm {...props} />
  );

  it('matches component snapshot', () => {
    const wrapper = pureComponent(fullProps);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
