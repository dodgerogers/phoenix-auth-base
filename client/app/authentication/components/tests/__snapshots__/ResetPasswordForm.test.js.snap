// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPasswordForm matches component snapshot 1`] = `
<div
  className="reset-password-form"
>
  <Form
    as="form"
    size="large"
  >
    <Field
      component={[Function]}
      fluid={true}
      name="resetPasswordToken"
      placeholder="Reset Password Token"
      validate={
        Array [
          [Function],
        ]
      }
    />
    <Field
      component={[Function]}
      fluid={true}
      icon="lock"
      iconPosition="left"
      name="password"
      placeholder="New Password"
      type="password"
      validate={
        Array [
          [Function],
          [Function],
        ]
      }
    />
    <Field
      component={[Function]}
      fluid={true}
      icon="lock"
      iconPosition="left"
      name="passwordConfirmation"
      placeholder="Password Confirmation"
      type="password"
      validate={
        Array [
          [Function],
          [Function],
        ]
      }
    />
    <Button
      as="button"
      color="teal"
      disabled={true}
      fluid={true}
      type="submit"
    >
      Submit
    </Button>
  </Form>
</div>
`;
